### PR TITLE
Adding the MAF Benchmark problems

### PR DESCRIPTION
**Pull request by IWNMWE**
Added the MAF 1-7 according to the following research paper : https://www.researchgate.net/publication/315446832_A_benchmark_test_suite_for_evolutionary_many-objective_optimization

### CODE REVIEW

**Comment by zoq** [comment]
Let me know if this is ready for review.

File: include/ensmallen_bits/problems/maf/maf4_function.hpp [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+ * Implementation of the fourth Maf test.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_PROBLEMS_MAF_THREE_FUNCTION_HPP
+#define ENSMALLEN_PROBLEMS_MAF_THREE_FUNCTION_HPP
```
Comment (rcurtin) [inline include/ensmallen_bits/problems/maf/maf4_function.hpp:14]: This is the same as in `maf4_function.hpp`, so this file won't ever actually be read.  Maybe can we make sure before we merge this PR that these functions are used in some test somewhere?

Comment (IWNMWE) [reply include/ensmallen_bits/problems/maf/maf4_function.hpp]: Sure will post some results this weekend with NSGA2 and AGEMOEA The consistent ones can be turned into tests

Comment (zoq) [reply include/ensmallen_bits/problems/maf/maf4_function.hpp]: Great!

**Comment by zoq** [comment]
@mlpack-jenkins test this please

**Comment by zoq** [comment]
If this comes back green, this should be ready to go.

**Review by zoq — APPROVED** [review]

**Comment by zoq** [comment]
Thanks for the contribution.

**Pull request merged**